### PR TITLE
Add perf model configs gb200

### PR DESCRIPTION
**Pull request by guyueh1**
> [!IMPORTANT]  
> The `Update branch` button must only be pressed in very rare occassions.
> An outdated branch is never blocking the merge of a PR.
> Please reach out to the automation team before pressing that button.

# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog

- Add specific line by line info of high level changes in this PR.

# Usage

- You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

**Comment by guyueh1** [comment]
@malay-nagda

File: scripts/llm/performance/recommended_model_configs/model_configs_gb200.csv [hunk 1]
```diff
@@ -0,0 +1,11 @@
+task,model,size,system,dtype,num_gpus,seq_len,tp_size,pp_size,cp_size,dp_size,ep_size,etp_size,vp_size,mbs,gbs,etp_size
```
Comment (malay-nagda) [inline scripts/llm/performance/recommended_model_configs/model_configs_gb200.csv:1]: `etp_size` is repeated twice after `ep_size` and `gbs`; will cause erroneous config parsing

File: scripts/llm/performance/pretrain_mixtral_8x22b.py [hunk 2]
```diff
@@ -66,6 +66,10 @@ def override_recipe_configs(
         recipe.trainer.plugins = bf16_with_fp8_mixed()
         recipe.trainer.plugins.grad_reduce_in_fp32 = False
 
+    # to mitigate the incorrect gradient_scaling_factor calculation in megatron.core
+    # under scenario average_in_collective=True and etp_size>1, disabling average_in_collective.
```
Comment (malay-nagda) [inline scripts/llm/performance/pretrain_mixtral_8x22b.py:70]: nitpick: `etp_size >= 1` as we want to set `average_in_collective=False` and `etp_size=1`

File: scripts/llm/performance/recommended_model_configs/model_configs_gb200.csv (hunk 1 above)
Comment (malay-nagda) [reply scripts/llm/performance/recommended_model_configs/model_configs_gb200.csv:1]: maybe we keep the one at the end to be consisted with H100 and B200 csv files and keep on adding new columns to the end as and when we add a new param?

Comment (guyueh1) [reply scripts/llm/performance/recommended_model_configs/model_configs_gb200.csv]: kept the etp_size column in the last, remove the other one

File: scripts/llm/performance/pretrain_mixtral_8x22b.py (hunk 2 above)
Comment (guyueh1) [reply scripts/llm/performance/pretrain_mixtral_8x22b.py]: changed it to: `if etp_size is not None and etp_size != tp_size`, as I made a typo, the real condition to trigger the gradient scaling error is when ETP!=TP

File: scripts/llm/performance/finetune_llama31_405b.py [hunk 3]
```diff
@@ -77,7 +77,7 @@ def override_recipe_configs(
         recipe.trainer.plugins = bf16_with_fp8_mixed()
         recipe.trainer.plugins.grad_reduce_in_fp32 = False
 
-    enable_cuda_graph = bool(args.gpu.lower() in [])
+    enable_cuda_graph = bool(args.gpu.lower() in ["gb200"])
```
Comment (guyueh1) [inline scripts/llm/performance/finetune_llama31_405b.py:80]: any particular reason we aren't enabling it for b200? I think it should provide benefit
@malay-nagda

Comment (malay-nagda) [reply scripts/llm/performance/finetune_llama31_405b.py:80]: I've updated it here- https://github.com/NVIDIA/NeMo/pull/12138...some things in my PR will need to be updated based on this PR...will commit once this PR is merged

**Review by malay-nagda — APPROVED** [review]
LGTM
